src/Mod/Cloud/App/CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ set(Cloud_LIBS
     FreeCADApp
     ${OPENSSL_LIBRARIES}
     ${CURL_LIBRARIES}
-    ${XercesC_LIBRARIES}
+    optimized ${XercesC_LIBRARIES} debug ${XercesC_LIBRARIE_DEBUG}
```

Comment:
Do we even need this line? Because FreeCADBase_LIBS is already set to use xerces debug & release plus boost and zlib we can actually drop ${Boost_LIBRARIES} and ${ZLIB_LIBRARIES} here, too.

Reply:
Just tried again without this line. Then  the debug build is linked against the release version.

Reply:
I just tested your PR and it exactly happens what I have feared. For a debug build under Linux cmake stops with an error:
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
XercesC_LIBRARIE_DEBUG
    linked by target "Cloud" in directory /home/test/FreeCAD/src/Mod/Cloud/App

This is because a separate debug version of xerces doesn't exist under Linux and that's why XercesC_LIBRARIE_DEBUG is empty.

Btw, did you realize the typo in XercesC_LIBRARIE_DEBUG? Either name it XercesC_LIBRARIES_DEBUG or better XercesC_LIBRARY_DEBUG.

Apart from this in src/Base/CMakeLists.txt we have this:
```
 if(MSVC)
    set(FreeCADBase_LIBS
        ${Boost_LIBRARIES}
        debug ${XercesC_DEBUG_LIBRARIES}
        optimized ${XercesC_LIBRARIES}
        ${ZLIB_LIBRARIES}
 ...
    )

```
i.e. the variable for the debug version is set to XercesC_DEBUG_LIBRARIES.

Then there is also an official FindXercesC.cmake that is part of CMake and there the variables XercesC_LIBRARY_RELEASE and XercesC_LIBRARY_DEBUG are used and from both XercesC_LIBRARIES is created.

If we don't switch to the official Xerces search module we should at least harmonize our implementation